app/models/grda_warehouse/warehouse_clients_processed.rb

```diff
@@ -810,6 +812,62 @@ def vispdat_priority_score(client_id)
       end
     end
 
+    def last_intentional_contacts(client_id)
+      open_enrollments = GrdaWarehouse::ServiceHistoryEnrollment.entry.ongoing.
+        where(client_id: @client_ids)
+      @services ||= {}.tap do |h|
+        open_enrollments.
+          joins(:service_history_services, :project).
```

Comment:
This is definitely going to need to know the max date, otherwise you're going to get all the services for the enrollment, which is one per-day for entry/exit.  I think you want to do something like:

```
max_date = open_enrollments.joins(:service_history_services).max(:date)
```
then add
```
where(date: max_date)
```
to the SHS query.

Reply:
Actually, I think you want to determine all three max dates, and then only pull records that occurred on the max of those 3 dates.

Reply:
* For NbN - use Services
* All open enrollments - use CurrentLivingSituation
* All open enrollment - use Event
* If driver Include CustomServices

For all of these try group by client_id, project_name, project_id and get maximum date